### PR TITLE
CI: replace golint with revive and run ineffassign in subdirectories 

### PR DESCRIPTION
**Pull request by RiSKeD**
I've replaced golint with revive, which is a drop-in replacement, because golint is no longer being maintained (see Issue #2522).
I also updated the call to the  [ineffassign tool ](https://github.com/gordonklaus/ineffassign) to include all subdirectories & fixed any ineffective assignment pointed out by the tool.

### CODE REVIEW

**Comment by RiSKeD** [comment]
Although I want to point out: that some of the fixed assignments are errors, never being checked afterwards...

File: cmds/exp/netbootxyz/netbootxyz.go [hunk 1]
```diff
@@ -175,7 +175,7 @@ func (o OSEndpoint) Load() error {
 			fmt.Printf("With Initrd at %s\n", tmpPath+"initrd")
 			fmt.Printf("Commandline: %s\n", o.Commandline)
 			// Load Kernel and initrd
-			err = kexec.FileLoad(vmlinuz, initrd, o.Commandline)
+			kexec.FileLoad(vmlinuz, initrd, o.Commandline)
 			// Load KExec kernel and initrd - init cmdline
```
Comment (rminnich) [inline cmds/exp/netbootxyz/netbootxyz.go:179]: this code is kind of crazy anyway. It uses the kexec package and ignores the error. That ought to get fixed, not ignored :-)

File: cmds/exp/netbootxyz/netbootxyz.go [hunk 2]
```diff
@@ -175,7 +175,7 @@ func (o OSEndpoint) Load() error {
...
 			cmd := exec.Command("kexec", "-e")
```
Comment (rminnich) [inline cmds/exp/netbootxyz/netbootxyz.go:180]: This is an odd bit of code. We have to be root at this point. so why not just do what kexec does?

                if err := kexec.Reboot(); err != nil {
                        log.Fatalf("%v", err)
it's a bit odd to run a command.

File: pkg/boot/fit/vfit.go [hunk 3]
```diff
@@ -103,7 +103,7 @@ func (s RSASignature) Verify(b []byte, ring openpgp.KeyRing) (*bytes.Reader, err
 		return r, err
 	}
 
-	hashed, err := vfile.CalculateHash(bytes.NewReader(b), s.hash.New())
+	hashed, _ := vfile.CalculateHash(bytes.NewReader(b), s.hash.New())
```
Comment (rminnich) [inline pkg/boot/fit/vfit.go:106]: we either need to stop ignoring an error or at least document why we feel it is safe to ignore the error.

File: pkg/boot/uefi/uefi.go [hunk 4]
```diff
@@ -73,7 +73,7 @@ func checkFVAndGetEntryPoint(name string) (uintptr, error) {
 	if err != nil {
 		return 0, err
 	}
-	secEntry, err := findSecurityCorePEEntry(dat)
+	secEntry, _ := findSecurityCorePEEntry(dat)
```
Comment (rminnich) [inline pkg/boot/uefi/uefi.go:76]: same here.

**Comment by rminnich** [comment]
is revive preferred to StaticCheck?

**Comment by RiSKeD** [comment]
Staticcheck does provide more rules than revive but switching to it will come with many edits across the project depending on the rules we want to check for. Therefore I would prefer to replace golint with revive for the moment as their rules set is almost identical in behaviour.

**Comment by RiSKeD** [comment]
@rminnich  Regarding the error checking: Can we mark those with a TODO and open up another PR or issue for fixing those? I would like to separate cleaning up the CI and potentially changing code behavior.

File: pkg/boot/uefi/uefi.go (hunk 4 above)
Comment (RiSKeD) [reply pkg/boot/uefi/uefi.go:78]: see #2539

File: pkg/boot/fit/vfit.go (hunk 3 above)
Comment (RiSKeD) [reply pkg/boot/fit/vfit.go:107]: see #2529

File: cmds/exp/netbootxyz/netbootxyz.go (hunk 1 above)
Comment (RiSKeD) [reply cmds/exp/netbootxyz/netbootxyz.go:180]: see #2529

File: cmds/exp/netbootxyz/netbootxyz.go (hunk 2 above)
Comment (RiSKeD) [reply cmds/exp/netbootxyz/netbootxyz.go:181]: I agree, using the kexec pkg is the way to go. see #2539

**Comment by RiSKeD** [comment]
Will rebase and revert the TODOs as soon as #2539 is merged.

**Review by rminnich — COMMENTED** [review]
Looks like you'll need to rebase and fix the conflicts, then we should be ok.

**Review by rminnich — APPROVED** [review]
These are all good cleanups now :-)

**Review by rminnich — APPROVED** [review]

**Pull request merged**